docs/proposals/202101_endpoint_discovery.md

```diff
@@ -42,20 +42,25 @@ info:
     blah: ...
 ```
 
-Right now this info API would be added to existing components Ruler, Store, Sidecar, Compact, Downsample, Receive. But this would be added in the future for Targets, Exemplers, etc.
+Right now this Info service would be added to existing components exposing a gRPC server.
+
+The upgrade path would be possible in this case, as it's strictly additive for endpoints, and there will be a migration path to fall back to current discovery methods which could be removed in the future.
+
+### Alternatives
 
-The upgrade path would be possible in this case, as it's strictly additive for endpoints, and there will be a migration path to fallback to current discovery methods which could be removed in the future.
+Initial discovery of gRPC services available will happen via gRPC reflection. To get further information, we would be adding an Info method to other services for example, Targets, Exemplars, etc. This would contain similar info or metadata information as we get right now from the Store Info method. We can always add more data to it in the future as needed.
 
-#### Alternatives
+This alternative solution was also considered, however it was not chosen because of its following drawbacks.
 
-Split Info method out of the Store API into a separate gRPC service called `info`. Each component must implement and expose this. The discovery would happen via this info gRPC service, which each component would need to explicitly state which API it supports. Besides that information, we would also need to include here various info, e.g. LabelSets, Max/Min time, etc. The drawback for this solution is harder upgrades as this would be more complex to implement as we would need to fall back to older APIs.
+1. For servers implementing multiple API (e.g. Store, Rules, Exemplar), we need to make multiple microservices calls (For given example 3x calls).
+2. There is also ambiguity about which Info method should be used for these kinds of servers.
 
 ### Upgrade plan
 
-Regardless of the solution we will be not removing any flags for a given period of time and have fallback code in place to be able to discover in the same way as we do right now. This will ensure smooth upgrade path for the users that make use of the current flags for discovery. We will mark the current flags as depracted and after two releases have passed we can remove the existing store and rule flag from query component and remove any migration code.
+Regardless of the solution, we will be not removing any flags for a given period and have a fallback code in place to be able to discover in the same way as we do right now. This will ensure a smooth upgrade path for the users that make use of the current flags for discovery. We will mark the current flags as deprecated and after two releases have passed we can remove the existing store and rule flag from the query component and remove any migration code.
```

Comment:
```suggestion
Regardless of the solution, we will be not removing any flags for a given period and have a fallback in place to be able to discover in the same way as we do right now. This will ensure a smooth upgrade path for the users who make use of the current flags for discovery. We will mark the current flags as deprecated and after two releases have passed we can remove the existing store and rule flag from the query component and remove any migration code.
```